cylc/flow/scripts/set.py

```diff
@@ -0,0 +1,378 @@
+#!/usr/bin/env python3
+
+# THIS FILE IS PART OF THE CYLC WORKFLOW ENGINE.
+# Copyright (C) NIWA & British Crown (Met Office) & Contributors.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""cylc set [OPTIONS] ARGS
+
+Set task prerequisites or outputs in a running workflow.
+
+Default: set all required outputs (note "succeeded" may be optional).
+
+Prerequisites:
+  Setting a task's prerequisites contributes to its readiness to run and
+  promotes it to the scheduler's active window. Use --pre=all to promote
+  even parentless tasks to the active window, where xtriggers become active.
```

Comment:
Your change doesn't quite work because we can't "set the prerequisites" of a parentless task - it doesn't have any prerequisites. Only `--pre=all` does that, because we (somewhat squiffily) interpret that as *set all prerequisites, if there are any, AND promote the task to the task pool*.

See also https://github.com/cylc/cylc-doc/issues/696 (regarding "active window")

Reply:
As far as I figure, my change improves the readability without changing the meaning? In which case, I am confused about what that sentence was trying to say.

This?
```suggestion
  parentless tasks to the active window. At this point it may still
  be waiting on any xtriggers.
```

Reply:
Sorry, I didn't read your change quite right. But still, the thing we need to represent is:
- setting any or all valid prerequisite promotes the target task to the task pool
- but parent-less tasks don't have any valid prerequisites, so the only way to promote them is `--set=all`

I'll consider the phrasing tomorrow.

Reply:
Done, see the latest CLI help text.